src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceUnitTest.kt

```diff
@@ -795,4 +800,86 @@ class ReferralServiceUnitTest {
     assertThat(response).isSameAs(sentReferral)
     verify(referralRepository).findByIdAndSentAtIsNotNull(sentReferral.id)
   }
+
+  @Nested
+  inner class UpdateDraftReferralDetails {
+    @BeforeEach
+    fun setup() {
+      whenever(referralRepository.save(any())).thenAnswer { it.arguments[0] }
+    }
+
+    @Test
+    fun `new referral details have the same created time as the referral`() {
+      val draftReferral = referralFactory.createDraft()
+
+      // there is no existing referral details for this referral id
+      whenever(referralDetailsRepository.findLatestByReferralId(draftReferral.id)).thenReturn(null)
+
+      referralService.updateDraftReferral(
+        draftReferral,
+        DraftReferralDTO(
+          completionDeadline = LocalDate.of(3022, 3, 28)
+        )
+      )
+
+      val captor = ArgumentCaptor.forClass(ReferralDetails::class.java)
+      verify(referralDetailsRepository).saveAndFlush(captor.capture())
+
+      val referralDetails = captor.value
+      assertThat(referralDetails.createdAt).isEqualTo(draftReferral.createdAt)
+    }
+
+    @Test
+    fun `updates to the referral details for draft referrals do not create new versions`() {
+      val draftReferral = referralFactory.createDraft()
+      val existingDetails = ReferralDetails(
+        UUID.randomUUID(),
+        null,
+        draftReferral.id,
+        draftReferral.createdAt,
+        draftReferral.createdBy.id,
+        "initial version",
+        LocalDate.of(2022, 3, 28),
+        null,
+        null,
+      )
+
+      whenever(referralDetailsRepository.findLatestByReferralId(draftReferral.id)).thenReturn(existingDetails)
+      referralService.updateDraftReferral(draftReferral, DraftReferralDTO(furtherInformation = "nothing to see here"))
```

Comment:
This `furtherInformation` field isn't used below, do we need it?

Reply:
the update has to have something in it otherwise it's not applied